lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -192,6 +192,27 @@ def retrying_submit(actions)
       end
     end
 
+    # we shouldn't send unsupported actions to ES
+    # method filters out unsupported actions by warning, writes event to DQL if enabled
+    # @returns valid actions
+    def filter_unsupported_actions(actions)
+      return if actions.nil? || actions.size < 1
```

Comment:
It is impossible for `retrying_submit` to receive `nil` for its `actions`, and therefore it is impossible for this method to receive `nil`. We can count on having an `Enumerable` (result of an upstream `Enumerable#map`) and can avoid the excess nil-checks.


Since we are relying on the return value being the actions that we should submit, this guard clause changes a given `Array` that is _empty_ into a `nil`. This is effectively handled somewhat accidentally in a guard clause in `retrying_submit`, but we can avoid the complexity because `Enumerable#partition` on an empty input array is _very_ fast so we really don't need to guard at all.

```suggestion
```